api/src/main/java/org/openmrs/module/oauth2/api/smart/impl/SmartAppManagementServiceImpl.java

```diff
@@ -0,0 +1,51 @@
+package org.openmrs.module.oauth2.api.smart.impl;
+
+import org.openmrs.api.impl.BaseOpenmrsService;
+import org.openmrs.module.oauth2.api.smart.SmartAppManagementService;
+import org.openmrs.module.oauth2.api.smart.db.hibernate.SmartAppDAO;
+import org.openmrs.module.oauth2.api.smart.model.SmartApp;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.transaction.annotation.Transactional;
+
+public class SmartAppManagementServiceImpl extends BaseOpenmrsService implements SmartAppManagementService {
+
+	@Autowired
+	SmartAppDAO smartAppDAO;
+
+	@Transactional
+	@Override
+	public SmartApp saveOrUpdateSmartApp(SmartApp smartApp) {
+		return smartAppDAO.saveOrUpdate(smartApp);
+	}
+
+	@Transactional
+	@Override
+	public SmartApp updateSmartApp(SmartApp smartApp) {
+		return smartAppDAO.update(smartApp);
+	}
+
+	@Transactional(readOnly = true)
+	@Override
+	public SmartApp getSmartApp(Integer smartId) {
+		return smartAppDAO.getById(smartId);
+	}
+
+	@Transactional(readOnly = true)
+	@Override
+	public SmartApp loadSmartAppByClientId(Integer clientId) {
+		return smartAppDAO.loadSmartAppByClientId(clientId);
+	}
+
+	@Transactional
+	@Override
+	public SmartApp unregisterSmartApp(SmartApp smartApp) {
+		smartAppDAO.delete(smartApp);
+		return smartApp;
+	}
+
+	@Transactional
+	@Override
+	public SmartApp merge(SmartApp smartApp) {
+		return smartAppDAO.merge(smartApp);
+	}
```

Comment:
methods to return SmartApp by clientDeveloper?

Reply:
Yeah... Will add it...  Just slipped from mind..